eclipse-run: <executionEnvironement> setting has no effect on dependency resolution

the docu of `tycho-eclipserun-plugin` states that setting `<executionEnvironement>` affects both dependency resolution and execution of the actual eclipse instance.

However, the dependency resolution part does not.

Steps to reproduce:


Execute `mvn verify` on the following `pom.xml` using **Java 11 as the system default JDK**:

```
<project
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0">
  <modelVersion>4.0.0</modelVersion>
  <groupId>foo.bar</groupId>
  <artifactId>foo.bar.baz</artifactId>
  <version>0.1.0-SNAPSHOT</version>
  <packaging>pom</packaging>


  <build>
    <plugins>
      <plugin>
        <groupId>org.eclipse.tycho.extras</groupId>
        <artifactId>tycho-eclipserun-plugin</artifactId>
        <version>2.7.5</version>
        <executions>
          <execution>
            <id>run-director</id>
            <goals>
              <goal>eclipse-run</goal>
            </goals>
            <phase>package</phase>
            <configuration>
              <applicationsArgs>
                <args>-consoleLog</args>
                <args>-application</args>
                <args>org.eclipse.equinox.p2.director</args>
                <args>-help</args>
              </applicationsArgs>
              <repositories>
                <repository>
                  <layout>p2</layout>
                  <url>mvn:org.eclipse.download.releasetrain:2022-09:202209141001:zip</url>
                </repository>
              </repositories>
              <dependencies>
                <dependency>
                  <artifactId>org.apache.felix.scr</artifactId>
                  <type>p2-installable-unit</type>
                </dependency>
                <dependency>
                  <artifactId>org.eclipse.equinox.p2.director.app</artifactId>
                  <type>p2-installable-unit</type>
                </dependency>
                <dependency>
                  <artifactId>org.eclipse.equinox.p2.core.feature.feature.group</artifactId>
                  <type>p2-installable-unit</type>
                </dependency>
                <dependency>
                  <artifactId>org.eclipse.osgi.compatibility.state</artifactId>
                  <type>p2-installable-unit</type>
                </dependency>
                <dependency>
                  <artifactId>org.eclipse.m2e.core</artifactId><!-- requires JavaSE-17 -->
                  <type>p2-installable-unit</type>
                </dependency>
              </dependencies>
            </configuration>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>

</project>
```

```
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------< foo.bar:foo.bar.baz >-------------------------
[INFO] Building foo.bar.baz 0.1.0-SNAPSHOT
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- tycho-eclipserun-plugin:2.7.5:eclipse-run (run-director) @ foo.bar.baz ---
[INFO] Fetching p2.index from mvn:org.eclipse.download.releasetrain:2022-09:202209141001:zip/
[INFO] Adding repository mvn:org.eclipse.download.releasetrain:2022-09:202209141001:zip
[INFO] Fetching content.jar from mvn:org.eclipse.download.releasetrain:2022-09:202209141001:zip/
[ERROR] org.eclipse.m2e.core 2.0.3.20220904-1703 requires Execution Environment that matches (&(osgi.ee=JavaSE)(version=17)) but the current resolution context uses [a.jre.javase 11.0.0]
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.eclipse.m2e.core 2.0.3.20220904-1703
[ERROR]   Missing requirement: org.eclipse.m2e.core 2.0.3.20220904-1703 requires 'osgi.ee; (&(osgi.ee=JavaSE)(version=17))' but it could not be found
[ERROR]
[ERROR] See https://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.552 s
[INFO] Finished at: 2022-09-21T16:18:11+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-eclipserun-plugin:2.7.5:eclipse-run (run-director) on project foo.bar.baz: Execution run-director of goal org.eclipse.tycho.extras:tycho-eclipserun-plugin:2.7.5:eclipse-run failed: Cannot resolve dependencies of null: See log for details -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```

This seems to be a regression of 0eeb95d1e58f3c93ced88dfb0f6a76da33d9074d:

Before, `executionEnvironment` was passed to the stub implementation, after, `executionEnvironment` is simply ignored.

--- a/tycho-extras/tycho-eclipserun-plugin/src/main/java/org/eclipse/tycho/extras/eclipserun/EclipseRunMojo.java
+++ b/tycho-extras/tycho-eclipserun-plugin/src/main/java/org/eclipse/tycho/extras/eclipserun/EclipseRunMojo.java
@@ -320,6 +320,7 @@ public class EclipseRunMojo extends AbstractMojo {
 		}
 		ExecutionEnvironmentConfiguration eeConfiguration = new ExecutionEnvironmentConfigurationImpl(logger, false,
 				toolchainManager, session);
+		eeConfiguration.setProfileConfiguration(executionEnvironment, "tycho-eclipserun-plugin <executionEnvironment>");
 		TargetPlatform targetPlatform = resolverFactory.getTargetPlatformFactory().createTargetPlatform(tpConfiguration,
 				eeConfiguration, null);
 		P2Resolver resolver = resolverFactory.createResolver(new MavenLoggerAdapter(logger, false));

